backend/client.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type ClickHouseClient struct {
+	context backend.PluginContext
+	settings *DatasourceSettings
+}
+
+// TODO add https support
+func (client *ClickHouseClient) Query(query string) (*Response, error) {
+
+	onErr := func(err error) (*Response, error) {
+		backend.Logger.Error(fmt.Sprintf("clickhouse client query error: %w", err)
+		return nil, err
+	}
+
+	datasourceUrl, err := url.Parse(client.context.URL)
+	if err != nil {
+		return onErr(fmt.Errorf("unable to parse clickhouse dataSourceUrl: %w", err))
+	}
+
+	httpClient := &http.Client{}
+
+	req, err := http.NewRequest(
+		"POST",
+		dataSourceUrl,
+		bytes.NewBufferString(query))
+	if err != nil {
+		return onErr(err)
+	}
+
+	req.Header.Set("X-ClickHouse-User", client.settings.Username)
```

Comment:
This removed parsing for our other settings such as withcredentials/usePost vs use get etc. that needs to be added back in.